nntrainer/src/conv2d_layer.cpp

```diff
@@ -111,7 +114,17 @@ int Conv2DLayer::setOptimizer(Optimizer &opt) {
   if (status != ML_ERROR_NONE)
     return status;
 
-  return this->opt.initialize(dim, true);
+  std::vector<Tensor> list_d;
+  for (unsigned int i = 0; i < filter_size; ++i) {
+    for (unsigned int j = 0; j < 2; ++j) {
+      list_d.push_back(Tensor(dim));
+    }
+    for (unsigned int j = 0; j < 2; ++j) {
+      list_d.push_back(Tensor(1, 1, 1));
+    }
+  }
+
+  return this->opt.initialize(list_d, true);
```

Comment:
How about directly passing the `weights` from `Layer`. This will be available once you rebase.

Reply:
We need to fix some issues related with weights. There will be garbage when we set the weights. This will be done next PR.